### DISCUSSION TITLE
I want to ask what technology you use to package your GUI

### DISCUSSION THREAD

**Discussion by Jzow** [hannobraun/fornjot#1612] [Q&A] [answered]
As far as I know, there is a Tauri Create that can package UI

**Comment by hannobraun** [comment, marked as answer]
Hey @Jzow, thank you for your interest!

I don't understand what the word "package" means in this context, so the following might not actually answer your question. Fornjot is using [wgpu](https://crates.io/crates/wgpu) for graphics and [egui](https://crates.io/crates/egui) (with [egui-wgpu](https://crates.io/crates/egui-wgpu)) for the UI. We do not use Tauri.

If you have any more questions, feel free to ask!

**Comment by hannobraun** [comment]
Converting this issue to a discussion, since it doesn't track any actionable work.

**Comment by Jzow** [comment]
@hannobraun thanks, I'm very sorry that my English level is not very good, The answer I want is   [egui-wgpu](https://crates.io/crates/egui-wgpu) crate

**Comment by Jzow** [comment]
In addition, I want to talk to you in detail. I see your information is from Germany. The industrial level of Germany is stronger than that of China on the whole, and the industrial software (**CAX**) in our country is not very perfect.

I'm also starting a business. I want to find a **CAD** project as a starting point. I found your open source project, and I think it's very interesting. The only disadvantage is that there is no way to install it quickly. For some industrial designers, they don't know much about software, such as nix, they can't install it. I think you have also taken this into consideration.

I think I can use your open source project to mark you as the author, and we can work together to do something, so that once there are more open source donations for you, it is a good thing, and we both have opportunities.

**Reply by hannobraun** [reply]
That sounds exciting! I'm always happy to have more people involved with Fornjot. Please note though, that Fornjot is very early in its development. It still lacks many important features, and what it does have is very immature. If it's a good starting point for you though, that's great!


> The only disadvantage is that there is no way to install it quickly. For some industrial designers, they don't know much about software, such as nix, they can't install it.

Which platforms are you targeting? Are you aware that I publish [weekly releases with pre-compiled binaries](https://github.com/hannobraun/Fornjot/releases) for Linux, Windows, and macOS?

**Reply by Jzow** [reply]
> Which platforms are you targeting? Are you aware that I publish [weekly releases with pre-compiled binaries](https://github.com/hannobraun/Fornjot/releases) for Linux, Windows, and macOS?

Yes, I noticed that. I downloaded the binary file of the mac version today, but i don't know much about nix, so I will look at nix again, and I want to try to install it later.

> It still lacks many important features, and what it does have is very immature. If it's a good starting point for you though, that's great!

I will always pay attention to it, starting from the most basic, and it will succeed over time. 

Whether you have considered transferring it to your own open source organization, which may be better publicized to the outside world. I can also provide some help to see how you think about it.

**Reply by hannobraun** [reply]
> Yes, I noticed that. I downloaded the binary file of the mac version today, but i don't know much about nix, so I will look at nix again, and I want to try to install it later.

Starting the Mac binary should just work! If it doesn't that's a bug. It's easily possible that a Mac-specific bug managed to sneak in, since I don't have access to a Mac, and I'm not sure if anyone is regularly testing Fornjot on Mac these days. What happens, if you start the Mac binary?

Nix has nothing to do with that. The repository contains a Nix package, which anyone using [NixOS](https://nixos.org/) (or anyone who is using the Nix package manager on another platform) can use to install Fornjot. But that is completely optional and just one way to install it.


> I will always pay attention to it, starting from the most basic, and it will succeed over time.

I certainly hope so :smile: 


> Whether you have considered transferring it to your own open source organization, which may be better publicized to the outside world. I can also provide some help to see how you think about it.

I have considered that. So far I haven't done it, because I'm not convinced there's any benefit to it, at this point. An organization makes it less clear who is maintaining the repository, and until now there has been no need to add other maintainers (nor have there been candidates for that role).

**Reply by Jzow** [reply]
> Nix has nothing to do with that. The repository contains a Nix package, which anyone using [NixOS](https://nixos.org/) (or anyone who is using the Nix package manager on another platform) can use to install Fornjot. But that is completely optional and just one way to install it.

I will conduct an installation test later. I will record any problems and share them with you.

**Reply by hannobraun** [reply]
Thank you!

**Comment by Jzow** [comment]
Double-click directly to open the binary file and an error will be reported:

`Failed to open the document "fj app". Text encoding Unicode (UTF-8) is not applicable.`

**Reply by hannobraun** [reply]
Hmm, that sounds like macOS doesn't recognize the file as an executable file. The reason is probably that the executable flag is not set. I remember that there was a problem like that (for Linux too), but I don't recall if that was a limitation of GitHub, or if that generally doesn't work for downloading executables via HTTP.

Can you try setting the executable flag? In a terminal, `chmod a+x fj-app` should work (execute in the directory where the Fornjot binary is located, and replace `fj-app` with the actuall name of the binary). There are probably other ways to do this without a terminal, but I'm not familiar enough with macOS to know that.

There must be a better way to distribute macOS binaries, but I don't know. It's not a system I use or have access to.

**Comment by Jzow** [comment]
Thanks, execute `chmod a+x fj-app` command, It can be accessed normally. I can look at source code  to see if there is a solution
